### ISSUE TITLE
Feedback for docs/advanced-topics/message-system/rpc.md

### ISSUE THREAD

**Issue by SteenPetersen** [Unity-Technologies/com.unity.multiplayer.docs#1209]
In the screenshots code examples, in the Update Method you invoke PingRpc(); with no int variable passed in, how can this be?

this in particular confuses me:

```  
[Rpc(SendTo.Server)]
public void PingRpc(int pingCount, RpcParams rpcParams)
{
    // Here we use RpcParams for incoming purposes - fetching the sender ID the RPC came from
    // That sender ID can be passed in to the PongRpc to send this back to that client and ONLY that client

    PongRpc(pingCount, "PONG!", RpcTarget.Single(rpcParams.Receive.SenderClientId, RpcTargetUse.Temp));
}

[Rpc(SendTo.SpecifiedInParams)]
void PongRpc(int pingCount, string message, RpcParams rpcParams)
{
    // We do not use rpcParams within this method's body, but that is okay!
    // The params passed in are used by the generated code to ensure that this sends only
    // to the one client it should go to

    Debug.Log($"Received pong from server for ping {pingCount} and message {message}");
}

void Update()
{
    if (IsClient && Input.GetKeyDown(KeyCode.P))
    {
        PingRpc(); // no int (pingCount) passed in
    }
}
```

I think I understand from the documentation that you wouldn't need to pass in the RpcParams as you can use them for 'incoming purposes' but surely you need to pass in the int?

**Comment by SteenPetersen** [comment]
Actually something else in the documentation is quite confusing:

After listing the 'Compile-time targets' there is a note stating: _"While client-to-client RPCs are supported, it is important to note that there are no direct connections between clients. When sending a client-to-client RPC, the RPC is sent to the server, that acts as a proxy, and forwards the same RPC message to all destination clients."_

That's fine, makes sense to me however, the code underneath is as follows:

```
// what is the point of PingRpc? why not call PongRpc directly since it will send to .NotServer and do the same thing as both of these methods combined?
[Rpc(SendTo.Server)] 
public void PingRpc(int pingCount)
{
    // Server -> Clients because PongRpc sends to NotServer
    // Note: This will send to all clients.
    // Sending to the specific client that requested the pong will be discussed in the next section.
    PongRpc(pingCount, "PONG!");
}

[Rpc(SendTo.NotServer)]
void PongRpc(int pingCount, string message)
{
    Debug.Log($"Received pong from server for ping {pingCount} and message {message}");
}

void Update()
{
    if (IsClient && Input.GetKeyDown(KeyCode.P))
    {
        // Client -> Server because PingRpc sends to Server
        PingRpc(); // again no parameters passed in
    }
}
```

**Comment by ShadauxCat** [comment]
@SteenPetersen In your first comment, the documentation is indeed incorrect. We'll make sure to fix that.

In your second comment, the code is stubbed in to show an example of sending a message to a server and receiving a response. It is not necessarily a _useful_ example, so to speak, in the sense that no one would likely include this in their production code; it is over-simplified to show only what needs to be shown. Typically, a ping RPC is used to calculate round-trip time, or to check for timeouts; that logic is left out here to avoid confusing people. And typically, the pong RPC is sent to a single client as a response to that client's specific ping RPC; as the comment in PingRpc mentions, the documentation is saving that for the next section to avoid overloading with too much information at once.

You are correct that, as written, the client could indeed just send PongRpc and get the same result, but that isn't what the documentation is trying to show here. It's a bit of a contrived example, but the idea is just to show what a server RPC and a not-server RPC look like. Does that help?

@jabbacakes It might be worth trying to come up with a better example here.

**Comment by SteenPetersen** [comment]
Yes, that makes sense to me now that I have been working with it for a bit. Thanks for the answer :-)

**Comment by jabbacakes** [comment]
For now, I've added a commit in #1253 to explain that the example code is heavily simplified to display basic properties of RPCs. I'll make a Jira ticket internally to cover improving the example code in the future.

@ShadauxCat can you fix the initial example that's missing an int? Probs just faster if you do it directly and commit to the open `issue-fixes` branch.

**Comment by jabbacakes** [comment]
For the sake of getting these updates in, I've merged the linked PR. Will make a note to go back and update the sample.

**Comment by ShaneMain** [comment]
Am I mistaken or has nothing been changed in regards to the initial issue here? Trying to find some sort of addendum on the docs lead me here. It's really confusing when you provide an example that doesn't even work to the extent that it contains a compiler error.
